### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by emyrue**
In this PR, we:
- Created the Solver class
- Created tests for Solver class
- Created factorial method
- Created reverse method
- Created fizzbuzz method

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+describe Solver do
+  context 'Testing solver methods' do
+    before :each do
+      @solver = Solver.new
+    end
+
+    it 'Factorial returns correct answer' do
+      expect(@solver.factorial(4)).to eq 24
+    end
```
Comment (omar-muhamad) [inline spec/solver_spec.rb:11]: - [ ] Please add all test cases for Getting the factorial of a number to cover all possibilities so add a test case for the factorial of a negative number and another one for the factorial of number 0 (zero). and then add those cases to your factorial class

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @emyrue and @dewslyse,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Github flow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by OLIPLICHE — APPROVED** [review]
Hi @emyrue, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
